File: 1-two-sum/two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int,vector<int>> hm;

        for (int i = 0; i < nums.size(); i++)
        {
            int num = nums[i];

            int diff = target - num;

            if (hm.find(diff) != hm.end()) {
                return {hm[diff].back(), i};
            }
            hm[num].push_back(i);
        }

        return {};
    }
};

File: 1-two-sum/two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        tracker = {}

        for i,_ in enumerate(nums):
            diff = target - _
            if diff in tracker:
                return [tracker.get(diff), i]
            tracker[_] = i
        
        return []
        